### PR TITLE
Setup quickstart RFS Docker Compose for local testing

### PR DESCRIPTION
**Pull request by lewijacn**
### Description
Changes were mainly focused on providing a quick and easy path for testing RFS in Docker. This included linking up the  created Dockerfile for ES with an S3 plugin to our Compose/Gradle infrastructure. Also added was the ability to preload a dataset onto the source cluster using an archive of the stored data. This allows for quicker testing of RFS by allowing a using to simply run`./gradlew composeUp -Pdataset='small-benchmark-single-node.tar.gz'` and have all containers needed, as well as a source with some data on it. Then a user can simply kick off RFS with a single command (as noted in the README) and see the RFS migration in progress. While an archive is currently stored in the `test-resources`, this isn't the path we want to take longer term for larger datasets which should be stored elsewhere, and would present issues for testing a cluster with more than a single node.

Also included are minor bug fixes from NPEs encountered when say no templates are configured or we are unable to find a Document id for a given Document (There is a task below raised for this as well)

### Issues Resolved
https://opensearch.atlassian.net/browse/MIGRATIONS-1601
https://opensearch.atlassian.net/browse/MIGRATIONS-1627

### Related Issues Opened
https://opensearch.atlassian.net/browse/MIGRATIONS-1649
https://opensearch.atlassian.net/browse/MIGRATIONS-1650

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: RFS/src/main/java/com/rfs/transformers/Transformer_ES_6_8_to_OS_2_11.java [hunk 1]
```diff
@@ -19,18 +19,20 @@ public ObjectNode transformGlobalMetadata(ObjectNode root) {
...
-            logger.info("Transforming template: " + templateName);
-            logger.debug("Original template: " + template.toString());
-            TransformFunctions.removeIntermediateMappingsLevels(template);
-            TransformFunctions.removeIntermediateIndexSettingsLevel(template); // run before fixNumberOfReplicas
-            TransformFunctions.fixReplicasForDimensionality(template, awarenessAttributeDimensionality);
-            logger.debug("Transformed template: " + template.toString());
-            templatesRoot.set(templateName, template);
-        });
-        newRoot.set("templates", templatesRoot);
+        if (root.get("templates") != null) {
```
[the remark this answers is not in the document]
Comment (peternied) [reply RFS/src/main/java/com/rfs/transformers/Transformer_ES_6_8_to_OS_2_11.java:22]: Sure - is there something tracking this 'future refactor' might be useful to know what that means / or the scope of expected changes.

**Review by peternied — APPROVED** [review]

File: RFS/build.gradle [hunk 2]
```diff
@@ -56,6 +61,26 @@ task copyDockerRuntimeJars (type: Copy) {
...
+        "reindexFromSnapshot": "reindex_from_snapshot",
+        "elasticsearchRFSSource": "elasticsearch_rfs_source"
+]
+
+dockerFilesInternalServices.forEach { projectName, dockerImageName ->
+    task "buildDockerImage_${projectName}" (type: DockerBuildImage) {
+        if (projectName == "reindexFromSnapshot") {
+            dependsOn copyDockerRuntimeJars
+            inputDir = project.file("./docker")
+        }
```
[the remark this answers is not in the document]
Comment (peternied) [reply RFS/build.gradle]: I like the change you made, looks better.  Can always iterate if we find bugs or need to expand functionality later.

File: RFS/docker/TestSource_ES_7_10/test-resources/inventory.md
Comment (peternied) [inline RFS/docker/TestSource_ES_7_10/test-resources/inventory.md:1]: 🔥 Nicely done!

**Comment by lewijacn** [comment]
As a note for this PR, the subsequent PR that builds upon this one refactors the preloaded data a bit so that we are no longer using any archive files in the repo and rather generating the datasets as an optional stage in the now multi-stage Dockerfile

File: RFS/docker/TestSource_ES_7_10/test-resources/inventory.md
Comment (lewijacn) [reply RFS/docker/TestSource_ES_7_10/test-resources/inventory.md:1]: Thanks, its a shame its life was so short though 😄

**Review by sumobrian — APPROVED** [review]

**Pull request merged**